model/riscv_ext_regs.sail

```diff
@@ -83,13 +83,17 @@ function ext_rvfi_init () = {
   x1 = x1 // to avoid hook being optimized out
 }
 
-
 /*!
 THIS(csrno, priv, isWrite) allows an extension to block access to csrno,
 at Privilege level priv. It should return true if the access is allowed.
 */
 val ext_check_CSR : (bits(12), Privilege, bool) -> bool
-function ext_check_CSR (csrno, p, isWrite) = true
+scattered function ext_check_CSR
```

Comment:
I understand I may have done something silly here so I'm explicitly asking for feedback. It felt like extensions needed to be able to add clauses to this function ad-hoc (Zkr being one of them), hence I've tried to add this ability in by turning it into a scattered function. See the `riscv_zkr.sail` file for how it get's used to check if access to the `seed` CSR is allowed.

Reply:
The various ext_* functions are for out-of-tree extensions like CHERI-RISC-V that need to override them with their own hooks, not for standard in-tree extensions.

Reply:
_Oh_. So, actually I need to revert this change, and also change the functions which the Zkr implementation uses then, since they all use `ext_*` functions as well. I assumed `ext` meant Extension in the RISC-V sense, not in an out of tree sense.

Reply:
Hmm okay. So, `seed` is a weird CSR in that _just_ reading it causes an Illegal Opcode Exception. One has to write to it in order to get the read value, and the written value is thrown away / ignored. This was to avoid having a standardised CSR which has side-effects on a read.

It's not obvious how this can be supported given how the current `execute CSR` function clause is written, unless I'm missing something? I can probably make the modifications as needed, but they look quite invasive to me.